kubernetes/samples/scripts/domain-lifecycle/stopServer.sh

```diff
@@ -0,0 +1,233 @@
+#   by default. If option to keep replica count constant ('-k') is selected, it only sets the 
+#   start policy to NEVER.
+# 
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+if [ "${debug}" == "true" ]; then set -x; fi;
+set -eu
+
+function usage() {
+
+  cat << EOF
+
+  This script stops a running WebLogic managed server in a domain either by
+  decreasing the value of 'spec.clusters[<cluster-name>].replicas' or by updating 
+  'spec.managedServers[<server-name>].serverStartPolicy' attribute of the domain 
+  resource or both as necessary. The 'spec.clusters[<cluster-name>].replicas' value
+  can be kept constant by using '-k' option. Please see README.md for more details.
+ 
+  Usage:
+ 
+    $(basename $0) -s myserver [-n mynamespace] [-d mydomainuid] [-k] [-m kubecli] [-v]
+  
+    -s <server_name>           : Server name parameter is required.
+
+    -d <domain_uid>            : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>             : Domain namespace. Default is 'sample-domain1-ns'.
+    
+    -k <keep_replica_constant> : Keep replica count constant. Default behavior is to decrement replica count.
+
+    -m <kubernetes_cli>        : Kubernetes command line interface. Default is 'kubectl' if KUBERNETES_CLI env
+                                 variable is not set. Otherwise default is the value of KUBERNETES_CLI env variable.
+
+    -v <verbose_mode>          : Enables verbose mode. Default is 'false'.
+
+    -h                         : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+serverName=""
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+keepReplicaConstant=false
+verboseMode=false
+serverStartPolicy=NEVER
+serverStarted=""
+action=""
+effectivePolicy=""
+managedServerPolicy=""
+stoppedWhenAlwaysPolicyReset=""
+withReplicas="CONSTANT"
+withPolicy="CONSTANT"
+
+while getopts "vks:m:n:d:h" opt; do
+  case $opt in
+    s) serverName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    k) keepReplicaConstant=true;
+    ;;
+    v) verboseMode=true;
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  # Validate that server name parameter is specified.
+  if [ -z "${serverName}" ]; then
+    validationError "Please specify the server name using '-s' parameter e.g. '-s managed-server1'."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the cluster name for current server 
+clusterName=$(${kubernetesCli} get pod ${domainUid}-${serverName} -n ${domainNamespace} -o=jsonpath="{.metadata.labels['weblogic\.clusterName']}")
```

Comment:
This depends on the pod being up, which isn't necessarily required as the pod may be temporarily down but still in a 'started' state.   Could this be replaced with the 'validateServerAndFindCluster' helper method instead?